sveltejs/language-tools
# (feat) TypeScript plugin

Initially support
- rename (doesn't work for prop renames yet, I'm okay with that)
- diagnostics
- find references
- go to definition

This makes all files TSX hardcoded for now, it seems the TS server is okay with importing tsx into js.

To test this, add 
```
        "typescriptServerPlugins": [
            {
                "name": "typescript-svelte-plugin",
                "enableForWorkspaceTypeScriptVersions": true,
                "path": "typescript-plugin"
            }
        ],
```

to the `contributes` property in the vscode extension's `package.json`

In contrast to #842 this does not set `"languages": ["svelte"]` which means TS will not take over Svelte files. The advantage is that diagnostics, rename etc within Svelte files stays in the control of the language-server, and only inside TS/JS files Svelte support is added. The drawback is that you need to save file changes to disk for Svelte files in order for the TS server to notice it.

Next steps after merge:
- publish plugin to npm
- make it opt-in in the VS Code extension
- check if we can move some common stuff into a shared folder in the monorepo; maybe we need to publish another npm package

#580
#550
#342
#110

@jasonlyu123 would be great if you could take a look 😃
